diesel/src/sqlite/connection/statement_iterator.rs

```diff
@@ -12,16 +12,75 @@ pub struct StatementIterator<'stmt, 'query> {
     field_count: usize,
 }
 
+impl<'stmt, 'query> StatementIterator<'stmt, 'query> {
+    #[cold]
+    fn handle_duplicated_row_case(
+        outer_last_row: &mut Rc<RefCell<PrivateSqliteRow<'stmt, 'query>>>,
+        column_names: &mut Option<Rc<[Option<String>]>>,
+        field_count: usize,
+    ) -> Option<QueryResult<SqliteRow<'stmt, 'query>>> {
+        // We don't own the statement. There is another existing reference, likly because
+        // a user stored the row in some long time container before calling next another time
+        // In this case we copy out the current values into a temporary store and advance
+        // the statement iterator internally afterwards
+        let last_row = {
+            let mut last_row = match outer_last_row.try_borrow_mut() {
+                Ok(o) => o,
+                Err(_e) => {
+                    return Some(Err(crate::result::Error::DeserializationError(
+                                    "Failed to reborrow row. Try to release any `SqliteField` or `SqliteValue` \
+                                     that exists at this point"
+                                        .into(),
+                                )));
+                }
+            };
+            let last_row = &mut *last_row;
+            let duplicated = last_row.duplicate(column_names);
+            std::mem::replace(last_row, duplicated)
```

Comment:
Unrelated to this PR, I would've assume based solely on the method name that this is an `unsafe` method?

Reply:
[`std::mem::replace`](https://doc.rust-lang.org/std/mem/fn.replace.html) is actually a safe function. It replaces the value at the location of the first argument with the value from the second argument, while returning the old value.